Add kotd library

jtzhao opened the pull request:
Add "kernel of the day" library to help kotd testing

coolo commented on tests/kernel/install_kotd.pm:
```diff
@@ -0,0 +1,46 @@
+sub kotd_reboot {
+    type_string("reboot\n");
+    wait_boot;
+    reset_consoles();
```
wait_boot is already doing that - and it even selects the root-console if you pass textmode => 1 to it.

jtzhao replied:
Fixed.

richiejp commented:
It looks like it should run perfectly, but a verification run would be nice. For example, you could write a test module which just calls these functions and then checks `uname -r` with a regex.

jtzhao commented:
@richiejp can't find a working up-to-date openQA server. Every time I upgrade openQA server, it breaks.

richiejp commented:
I have not had any problems upgrading OpenQA with zypper. If you are using the os-autoinst source then you may need to recompile the C++ components. You can ask for help fixing any problems on IRC or on the mailing list.

foursixnine commented:
@jtzhao Can you define "the server breaks"? What's your current setup? (As @richiejp mentioned, you can ask for help on irc)

richiejp commented:
It seems that @pevik may have the same issue. It might just be because some shared libraries are outdated and you need to restart all the services (just restart the whole system).

jtzhao commented:
Removed kotd_reboot function because I can't get it working. We can reboot inside the actual test module though.
Test result: http://147.2.207.102/tests/630#step/test/21

okurz commented:
please [squash commits](https://robots.thoughtbot.com/git-interactive-rebase-squash-amend-rewriting-history)

verification looks ok but still the library is not used anywhere in the test distribution so I do not see the purpose of this module (yet).

jtzhao commented:
@okurz https://progress.opensuse.org/issues/15042

jtzhao commented:
I've converted this module into a test and added it into products/sle/main.pm. Now during ltp and regression testing, if INSTALL_KOTD is set(e.g. SLE12-SP3), kotd kernel will be installed ahead of other steps.
Test result: http://147.2.207.102/tests/648

richiejp commented:
Excellent.
